hs_core/management/commands/remove_bags_depending_on_last_download.py

```diff
@@ -0,0 +1,84 @@
+import math
+
+from datetime import timedelta
+from django.core.exceptions import ValidationError
+from django.core.management.base import BaseCommand
+from django.db.models import Q
+from django.utils import timezone
+from hs_core.models import BaseResource
+from hs_core.hydroshare.hs_bagit import delete_bag
+from hs_core.hydroshare.utils import set_dirty_bag_flag
+
+
+class Command(BaseCommand):
+    help = "Remove bags for resources that haven't been downloaded recently"
+
+    def add_arguments(self, parser):
+
+        parser.add_argument('--weeks', type=int, dest='weeks', help='include res not downloaded in the last X weeks')
+
+        parser.add_argument(
+            '--published',
+            action='store_true',  # True for presence, False for absence
+            dest='published',  # value is options['published']
+            help='filter to include only published resources')
+
+        parser.add_argument(
+            '--dryrun',
+            action='store_true',
+            dest='dryrun',
+            default=False,
+            help='Show cumulative size of bags to be removed without actually removing them',
+        )
+
+    def convert_size(self, size_bytes):
+        if size_bytes == 0:
+            return "0B"
+        size_name = ("B", "KB", "MB", "GB", "TB")
+        i = int(math.floor(math.log(size_bytes, 1024)))
+        p = math.pow(1024, i)
+        s = round(size_bytes / p, 2)
+        return "%s %s" % (s, size_name[i])
+
+    def handle(self, *args, **options):
+        resources = []
+        add_message = ""
+        weeks = options['weeks']
```

Comment:
Should we be checking for a minimum allowed value for weeks unless deleting all existing bags is not an issue?

Reply:
good call, thanks! added a min value